gfx/video_thread_wrapper.c

```diff
@@ -697,8 +710,6 @@ static bool video_thread_set_shader(void *data,
 {
    thread_packet_t pkt;
    thread_video_t *thr = (thread_video_t*)data;
-   if (!thr)
```

Comment:
All of these NULL checks need to be restored.

I cannot turn threaded rendering on with your PR, because NULL is passed on driver reinit (and RA subsequently segfaults). The specific violation in this case happens in `thread_get_flags()` - but all functions should be made 'safe', I think.

Reply:
That's odd, many of the functions already didn't include the sanity check so I assumed they were all redundant at that point because only init could have "thr" (and thus st_video->data) be NULL.

I guess I'll have to restore those and add them to the functions that didn't have it previously, for the sake of consistency.